server/qrops.go

```diff
@@ -0,0 +1,291 @@
+package server
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"image"
+	"net/url"
+	"strconv"
+
+	"github.com/yeqown/go-qrcode/v2"
+	"github.com/yeqown/go-qrcode/writer/standard"
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/images"
+	"github.com/status-im/status-go/multiaccounts"
+)
+
+type WriterCloserByteBuffer struct {
+	*bytes.Buffer
+}
+
+func (wc WriterCloserByteBuffer) Close() error {
+	return nil
+}
+
+func NewWriterCloserByteBuffer() *WriterCloserByteBuffer {
+	return &WriterCloserByteBuffer{bytes.NewBuffer([]byte{})}
+}
+
+type QRConfig struct {
+	DecodedQRURL    string
+	WithLogo        bool
+	CorrectionLevel qrcode.EncodeOption
+	KeyUID          string
+	ImageName       string
+	Size            int
+	Params          url.Values
+}
+
+func NewQRConfig(params url.Values, logger *zap.Logger) *QRConfig {
+	config := &QRConfig{}
+	config.Params = params
+	err := config.setQrURL()
+
+	if err != nil {
+		logger.Error("[qrops-error] error in setting QRURL", zap.Error(err))
+	}
+
+	config.setAllowProfileImage()
+	config.setErrorCorrectionLevel()
+	err = config.setSize()
+
+	if err != nil {
+		logger.Error("[qrops-error] could not convert string to int for size param ", zap.Error(err))
+	}
+
+	if config.WithLogo {
+		err = config.setKeyUID()
+
+		if err != nil {
+			logger.Error(err.Error())
+		}
```

Comment:
If a Key UID is required, what would affect would having no key UID have on later code execution?

Reply:
fixed as a side effect of : https://github.com/status-im/status-go/commit/8d1a34bcda651166d09ea2672711e0b662200d54